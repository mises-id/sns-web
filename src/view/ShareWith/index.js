/*
 * @Author: lmk
 * @Date: 2021-07-15 10:45:52
 * @LastEditTime: 2021-12-31 11:32:06
 * @LastEditors: lmk
 * @Description: create mises page
 */
import "./index.scss";
import React, { useEffect, useState } from "react";
import { NavBar, Picker } from "zarm";
import { useTranslation } from "react-i18next";
import radio_selected from "@/images/radio_selected.png";
import radioIcon from "@/images/radio.png";
import arrowDown from "@/images/arrow-down.png";
import { shareWith } from "@/utils";
const ShareWith = ({history={}}) => {
  const max = 9;
  let start = 1;
  const selectHrsOptions = [{
    label:0.5,
    value:0.5
  }];
  while (start<max) {
    selectHrsOptions.push({
      label:start,
      value:start
    })
    ++start;
  }
  const [selectHrs, setselectHrs] = useState(0.5)
  const { t } = useTranslation();
  const [select, setselect] = useState("public");
  const [pickerVisible, setpickerVisible] = useState(false);
  useEffect(() => {
    if(history.state&&history.state.selectType){
      setselect(history.state.selectType)
      setselectHrs(history.state.selectHrs)
    }
  }, []) // eslint-disable-line react-hooks/exhaustive-deps
  const RadioItem = ({ label, name, change}) => {
    const checked = select === name;
    return (
      <div onClick={() => change(name)} className="m-flex">
        <img
          src={checked ? radio_selected : radioIcon}
          alt=""
          width={20}
          height={20}
        />
        <div className="m-margin-left15">
          <span className="m-font17">{label}</span>
        </div>
      </div>
    );
  };
  const getChange = name=>{
    setselect(name)
  }
  const limited = ()=>{
    return <div className="limit m-flex" onClick={() => setpickerVisible(true)}>
      <span className="m-font14 m-colors-666">Turn to private after:</span>
      <div className="select m-flex m-font14 m-row-between">
        {selectHrs}
        <img src={arrowDown} alt="" width={11} className={`select-icon ${pickerVisible ? 'show-selectPicker' : ''}`}/>
      </div>
      <span className="m-font14 m-colors-666">hrs</span>
    </div>
  }
  const pickerSubmit = (e) => {
    if (e.length) {
      setselectHrs(e[0].value)
    }
    setpickerVisible(false);
  };
  const getSelect = ()=>{
    history.state = {
      ...history.state,
      selectType: select,
      selectHrs
    }
    window.history.back()
  }
  return (
    <div>
      <NavBar
        left={
          <span onClick={() => window.history.back()} className="m-font16">
            {t("cancel")}
          </span>
        }
        right={<span className="m-colors-5c65f6 m-font16" onClick={getSelect}>{t("done")}</span>}
      />
      <div className="m-flex m-flex-col m-layout">
        {shareWith.map((val, index) => (
          <div className="m-line-bottom radio-item" key={index} >
            <RadioItem label={val.label} name={val.value} change={getChange}/>
            {val.value==='limited'&&select==='limited'&&limited()}
          </div>
        ))}
      </div>
      <Picker
        visible={pickerVisible}
        dataSource={selectHrsOptions}
        defaultValue={selectHrs}
        onOk={pickerSubmit}
        onCancel={() => setpickerVisible(false)}
        wheelDefaultValue={selectHrs}
        value={selectHrs}
      ></Picker>
    </div>
  );
};
export default ShareWith;
