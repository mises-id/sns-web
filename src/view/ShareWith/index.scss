.radio-item{
  padding: 28px 0 28px 15px;
  background-color: white;
}
.limit{
  padding-left: 35px;
  line-height: 1;
  margin-top: 18px;
}
.select{
  width: 60px;
  height: 30px;
  border: 1px solid #EEEEEE;
  border-radius: 3px;
  line-height: 30px;
  padding-left: 15px;
  padding-right: 8px;
  margin-left: 15px;
  margin-right: 7px;
}
.show-selectPicker{
  transform: rotate(180deg);
}
.select-icon{
  transition: all 0.3s;
}