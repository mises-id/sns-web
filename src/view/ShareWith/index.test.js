/*
 * @Author: lmk
 * @Date: 2021-07-16 00:14:28
 * @LastEditTime: 2021-12-31 10:01:11
 * @LastEditors: lmk
 * @Description: MyPost test
 */

import React from 'react';
import ShareWith from './';
import {shallow } from 'enzyme';
describe('shareWith', () => {
  it('should render correctly', () => {
    const component = shallow(<ShareWith />)
    expect(component).toMatchSnapshot();
  });
});