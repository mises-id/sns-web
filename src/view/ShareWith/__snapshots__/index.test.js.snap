// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shareWith should render correctly 1`] = `
<div>
  <NavBar
    left={
      <span
        className="m-font16"
        onClick={[Function]}
      >
        cancel
      </span>
    }
    prefixCls="za-nav-bar"
    right={
      <span
        className="m-colors-5c65f6 m-font16"
        onClick={[Function]}
      >
        done
      </span>
    }
  />
  <div
    className="m-flex m-flex-col m-layout"
  >
    <div
      className="m-line-bottom radio-item"
      key="0"
    >
      <RadioItem
        change={[Function]}
        label="Public"
        name="public"
      />
    </div>
    <div
      className="m-line-bottom radio-item"
      key="1"
    >
      <RadioItem
        change={[Function]}
        label="Private"
        name="private"
      />
    </div>
    <div
      className="m-line-bottom radio-item"
      key="2"
    >
      <RadioItem
        change={[Function]}
        label="Time Limited Post"
        name="limited"
      />
    </div>
  </div>
  <ForwardRef(forwardRef)
    dataSource={
      Array [
        Object {
          "label": 0.5,
          "value": 0.5,
        },
        Object {
          "label": 1,
          "value": 1,
        },
        Object {
          "label": 2,
          "value": 2,
        },
        Object {
          "label": 3,
          "value": 3,
        },
        Object {
          "label": 4,
          "value": 4,
        },
        Object {
          "label": 5,
          "value": 5,
        },
        Object {
          "label": 6,
          "value": 6,
        },
        Object {
          "label": 7,
          "value": 7,
        },
        Object {
          "label": 8,
          "value": 8,
        },
      ]
    }
    defaultValue={0.5}
    onCancel={[Function]}
    onOk={[Function]}
    value={0.5}
    visible={false}
    wheelDefaultValue={0.5}
  />
</div>
`;
